### PR TITLE
Adds a helm chart, dockerfile, and instructions for running examples with Kubernetes

### PR DESCRIPTION
**Pull request by dmsuehir**
# What does this PR do?

This PR adds a helm chart, Dockerfile/docker compose, and instructions for running the optimum-habana examples using Kubernetes.

I added an `examples/kubernetes` directory, which is similar to `examples/multi-node-training` in that it's a general example that can be used with any of the other examples.

Since the docker images isn't published to a repository anywhere at this time, the user will need to build and push the container before deploying the helm chart. The Dockerfile uses the Habana Gaudi docker image and it's base and then installs optimum-habana, Habana's fork of DeepSpeed, and a `requirements.txt` file. The `requirements.txt` file is just installing the `requirements.txt` file from the other examples directories (like `language-modeling/requirements.txt`). The user can adjust this depending on which example they want to run.

I set up this example to be similar to the way the Intel AI Containers repo is setup with the README being generated using helm-doc, the dockerfile being built using docker-compose, and a `ci` folder with Helm values files for validated configurations.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/kubernetes/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        no_proxy: ""
+        BASE_IMAGE_NAME: ${BASE_IMAGE_NAME:-vault.habana.ai/gaudi-docker/1.15.1/ubuntu22.04/habanalabs/pytorch-installer-2.2.0}
+        BASE_IMAGE_TAG: ${BASE_IMAGE_TAG:-latest}
+        GAUDI_SW_VER: 1.15.1
+        OPTIMUM_HABANA_VER: 1.11.1
+        REGISTRY: ${REGISTRY}
+        REPO: ${REPO}
+      context: .
+      labels:
+        org.opencontainers.base.name: "vault.habana.ai/gaudi-docker/1.15.1/ubuntu22.04/habanalabs/pytorch-installer-2.2.0:latest"
```
Comment (ashahba) [inline examples/kubernetes/docker-compose.yaml]: I'd replace `1.15.1` with `${GAUDI_SW_VER:-1.15.1}` to stay consistent.

**Review by ashahba — COMMENTED** [review]
Thanks Dina!
only a minor comment.

File: examples/kubernetes/docker-compose.yaml (hunk 1 above)
Comment (dmsuehir) [reply examples/kubernetes/docker-compose.yaml]: Thanks, I updated it to use `BASE_IMAGE_NAME` and `BASE_IMAGE_TAG`

**Review by ashahba — APPROVED** [review]
LGTM!

**Comment by HarshaRamayanam** [comment]
LGTM

**Comment by sywangyi** [comment]
hi @dmsuehir 1.16 Driver and dockerimage has been released. you may need to upgrade to it. https://github.com/huggingface/optimum-habana/releases/tag/v1.12.0 is released to use 1.16

**Comment by dmsuehir** [comment]
PR has been posted to optimum-habana

**Pull request closed without merging**